RecoTracker/MkFitCore/src/TrackerInfo.cc

```diff
@@ -44,19 +72,192 @@ namespace mkfit {
     return l;
   }
 
-  LayerInfo &TrackerInfo::new_barrel_layer() {
+  LayerInfo& TrackerInfo::new_barrel_layer() {
     m_barrel.push_back(new_layer(LayerInfo::Barrel));
     return m_layers.back();
   }
 
-  LayerInfo &TrackerInfo::new_ecap_pos_layer() {
+  LayerInfo& TrackerInfo::new_ecap_pos_layer() {
     m_ecap_pos.push_back(new_layer(LayerInfo::EndCapPos));
     return m_layers.back();
   }
 
-  LayerInfo &TrackerInfo::new_ecap_neg_layer() {
+  LayerInfo& TrackerInfo::new_ecap_neg_layer() {
     m_ecap_neg.push_back(new_layer(LayerInfo::EndCapNeg));
     return m_layers.back();
   }
 
+  int TrackerInfo::n_total_modules() const {
+    int nm = 0;
+    for (auto& l : m_layers)
+      nm += l.n_modules();
+    return nm;
+  }
+
+  //==============================================================================
+
+  namespace {
+    struct GeomFileHeader {
+      int f_magic = s_magic;
+      int f_format_version = s_version;
+      int f_sizeof_trackerinfo = sizeof(TrackerInfo);
+      int f_sizeof_layerinfo = sizeof(LayerInfo);
+      int f_sizeof_moduleinfo = sizeof(ModuleInfo);
+      int f_n_layers = -1;
+
+      GeomFileHeader() = default;
+
+      constexpr static int s_magic = 0xB00F;
+      constexpr static int s_version = 1;
+    };
+
+    template <typename T>
+    int write_std_vec(FILE* fp, const std::vector<T>& vec, int el_size = 0) {
```

Comment:
can you actually unpack a bit the purpose of this binary file? Is it for private/standalone debugging, or something intended to be used in the official development process?
I could be a little concerned about the long-term maintainability in the latter case.

Reply:
This is for the geometry to be picked up in standalone.